### PR TITLE
@example tag

### PR DESCRIPTION
**Pull request by openback**
Added support for the `@example` tag, a la [phpDocumentor](http://www.phpdoc.org/).

### CODE REVIEW

File: src/org/salesforce/apexdoc/ApexDoc.java [hunk 1]
```diff
@@ -445,8 +446,25 @@ private static void fillMethodModel(MethodModel mModel, String name, ArrayList<S
                 continue;
             }
 
-            // handle multiple lines for description.
-            if (inDescription) {
+            idxStart = comment.toLowerCase().indexOf("@example");
+            if (idxStart != -1 || i == 1) {
+                if (idxStart != -1 && comment.length() >= idxStart + 10) {
```
Comment (davidhabib) [inline src/org/salesforce/apexdoc/ApexDoc.java]: shouldn't this be idxStart + 8?

File: src/org/salesforce/apexdoc/ApexDoc.java [hunk 2]
```diff
@@ -397,6 +397,7 @@ private static void fillPropertyModel(PropertyModel propertyModel, String name,
     private static void fillMethodModel(MethodModel mModel, String name, ArrayList<String> lstComments, int iLine) {
         mModel.setNameLine(name, iLine);
         boolean inDescription = false;
+        boolean inExample = false;
```
Comment (davidhabib) [inline src/org/salesforce/apexdoc/ApexDoc.java]: below inside the loop, for every line that sets inDescription = false when a new @token is found, should also set inExample = false?

File: src/org/salesforce/apexdoc/ApexDoc.java [hunk 3]
```diff
@@ -454,7 +472,18 @@ private static void fillMethodModel(MethodModel mModel, String name, ArrayList<S
                         break;
                 }
                 if (j < comment.length()) {
-                    mModel.setDescription(mModel.getDescription() + ' ' + comment.substring(j));
+                    if (inDescription) {
+                        mModel.setDescription(mModel.getDescription() + ' ' + comment.substring(j));
+                    } else if (inExample) {
```
Comment (davidhabib) [inline src/org/salesforce/apexdoc/ApexDoc.java]: why is the example code different than the description code?  are you trying to support @example repeated on multiple lines?  is that something javadoc does?

File: src/org/salesforce/apexdoc/ApexDoc.java (hunk 1 above)
Comment (openback) [inline src/org/salesforce/apexdoc/ApexDoc.java]: To be honest, without comments, I wasn't totally sure what this conditional was for, but mine seemed to work fine! Could you elaborate on this section?

File: src/org/salesforce/apexdoc/ApexDoc.java (hunk 2 above)
Comment (openback) [inline src/org/salesforce/apexdoc/ApexDoc.java]: Ahh, good point

File: src/org/salesforce/apexdoc/ApexDoc.java (hunk 3 above)
Comment (openback) [inline src/org/salesforce/apexdoc/ApexDoc.java]: This change is to preserve code indentation and new lines, which `@description` doesn't do.

File: src/org/salesforce/apexdoc/ApexDoc.java (hunk 2 above)
Comment (openback) [inline src/org/salesforce/apexdoc/ApexDoc.java]: And now dealt with.

**Pull request merged**